MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/analytics/AnalyticsTracker.java

```diff
@@ -0,0 +1,134 @@
+package com.mapbox.mapboxandroiddemo.analytics;
+
+
+import android.os.Build;
+import android.support.annotation.NonNull;
+
+import com.segment.analytics.Analytics;
+import com.segment.analytics.messages.IdentifyMessage;
+import com.segment.analytics.messages.ScreenMessage;
+import com.segment.analytics.messages.TrackMessage;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by langstonsmith on 4/25/17.
+ */
+
+public class AnalyticsTracker {
+
+  private static final AnalyticsTracker INSTANCE = new AnalyticsTracker();
+
+  private String openedApp = "Opened app";
+  private String clickedOnSignInButtonEventName = "Clicked on sign in button";
+  private String clickedOnCreateAccountButtonEventName = "Clicked on create account button";
+  private String clickedOnNavDrawerSectionEventName = "Clicked on nav drawer section";
+  private String clickedOnIndividualExampleEventName = "Clicked on individual example";
+  private String sectionNameMapKey = "section name";
+  private String exampleNameMapKey = "example name";
+  private String isTabletMapValue = "tablet";
+  private String isPhoneMapValue = "phone";
+
+
+  public static AnalyticsTracker get() {
```

Comment:
Apologies for missing out on this, I'm not a singleton expert. On one end it seems that this approach will be faster in comparison what I use (eg. http://www.androiddesignpatterns.com/2012/05/correctly-managing-your-sqlite-database.html), though it both setups should be thread safe. Note that faster is also relative as in how many occasions would you call getInstance at the same time from 2 threads when there isn't an instance yet?